example/hyperparam/gPyOpt/search_keras.py

```diff
@@ -0,0 +1,146 @@
+import math
+
+import os
+import shutil
+import tempfile
+
+import click
+import GPyOpt
+
+import mlflow
+import mlflow.sklearn
+import mlflow.tracking
+import mlflow.projects
+
+"""
+Example of hyper param search in MLflow using GPyOpt.
+"""
+
+@click.command(help="Perform hyper parameter search with GPyOpt library."
+                    "Optimize dl_train target.")
+@click.option("--max-runs", type=click.INT, default=100,
+              help="Maximum number of runs to evaluate.")
+@click.option("--batch-size", type=click.INT, default=4,
+              help="Number of runs to evaluate in a batch")
+@click.option("--max-p", type=click.INT, default=4,
+              help="Maximum number of parallel runs.")
+@click.option("--epochs", type=click.INT, default=500,
+              help="Number of epochs")
+@click.option("--metric", type=click.STRING, default="rmse",
+              help="Metric to optimize on.")
+@click.option("--gpy-model", type=click.STRING, default="GP_MCMC",
+              help="Optimizer algorhitm.")
+@click.option("--gpy-acquisition", type=click.STRING, default="EI_MCMC",
+              help="Optimizer algorhitm.")
+@click.option("--initial-design", type=click.STRING, default="random",
+              help="Optimizer algorhitm.")
+@click.option("--seed", type=click.INT, default=97531,
+              help="Seed for the random generator")
+@click.option("--training-experiment-id", type=click.INT, default=-1,
+              help="Maximum number of runs to evaluate. Inherit parent;s experiment if == -1.")
+@click.argument("training_data")
+def run(training_data, max_runs, batch_size, max_p, epochs, metric, gpy_model, gpy_acquisition,
+        initial_design, seed, training_experiment_id):
+    bounds = [
+        {'name': 'lr', 'type': 'continuous', 'domain': (1e-3, 1e-1)},
+        {'name': 'lr_decay', 'type': 'continuous', 'domain': (.8, 1)},
+        {'name': 'rho', 'type': 'continuous', 'domain': (.8, 1)},
+    ]
+    # create random file to store run ids of the training tasks
+    tmp = tempfile.mkdtemp()
+    results_path = os.path.join(tmp, "results.txt")
+
+    def eval(parms):
+        """
+        Train keras model with given parameters by invoking mlflow run.
+
+        Notice we store runUuid and resulting metric in a file. We will later use these to pick the
+        best run and to log the runUuids of the child runs as an artifact.
```

Comment:
Maybe we can write that this is a temporary workaround until we can add tags to our MLflow runs to search for them directly.